### ISSUE TITLE
False Positive $var === null || $var !== null seems to be always true

### ISSUE THREAD

**Issue by MaximilianKresse** [kalessil/phpinspectionsea#1620]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), v3.0.10 |
| Language level | PHP 7.3                       |

Current behaviour
--
```php
function test(string $name, ?string $beforeName): bool {
    return (
        ($beforeName === null && isFree($name))
        || ($beforeName !== null && isBefore($name, $beforeName))
    );
}

```

![image](https://user-images.githubusercontent.com/545671/90792950-68c35a80-e30b-11ea-9ba6-edf8f93201f9.png)

Expected behaviour
--
No errors :-)

**Comment by ea-inspections-team** [comment]
Thanks for reporting @MaximilianKresse!

It's indeed `$beforeName !== null` true at that point. The condition can be simplified to `($beforeName === null && isFree($name)) || isBefore($name, $beforeName)`. 

But the message definitely needs to be improved, as the code fragment in the message should explain why we are complaining. We'll improve it!

**Comment by ea-inspections-team** [comment]
Update: examined more carefully, understand the issue now. We'll look into it.

**Comment by ea-inspections-team** [comment]
Thank you for reporting @MaximilianKresse! Fixed!
